### PR TITLE
added modal component/used in register for terms

### PR DESCRIPTION
**Pull request by Hanya-Ahmad**
### Description
Created a modal component and used it for Terms and Privacy Policy modals. 
### Issues
- #78 
### Design
image
image

### CODE REVIEW

File: frontend/src/components/ModalComponent.vue [hunk 1]
```diff
@@ -0,0 +1,65 @@
+<template>
+  <v-dialog
+    v-model="internalDialog"
+    @update:modelValue="dialogUpdated"
+    max-width="50%"
+  >
+    <v-card>
+      <v-card-title class="headline">
+        <slot name="title" class="title">Modal Title</slot>
```
Comment (mahmoud-emad) [inline frontend/src/components/ModalComponent.vue]: Shall we remove the `Modal Title`?

File: frontend/src/components/ModalComponent.vue [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      </v-card-title>
+      <v-card-text>
+        <slot></slot>
```
Comment (mahmoud-emad) [inline frontend/src/components/ModalComponent.vue]: Give this slot a name like `modal body`

File: frontend/src/components/ModalComponent.vue [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      </v-card-text>
+      <v-card-actions>
+        <v-spacer></v-spacer>
+
+        <slot name="custom-button"></slot>
+        <v-btn class="btn" @click="closeModal">Close</v-btn>
```
Comment (mahmoud-emad) [inline frontend/src/components/ModalComponent.vue]: This slot should containes the the two of buttons

File: frontend/src/components/TermsModalComponent.vue [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<template>
+  <div>
+    <v-btn @click="$refs.dialog = true">Show Terms and Conditions</v-btn>
+
+    <!-- Embed the BasicModal component and customize its content -->
+    <ModalComponent ref="termsModal">
+      <template v-slot:title> Terms and Conditions </template>
+      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit...</p>
+      <!-- Add more content here as needed -->
+    </ModalComponent>
```
Comment (mahmoud-emad) [inline frontend/src/components/TermsModalComponent.vue]: Also here we should add the buttons

**Review by Mahmoud-Emad — CHANGES_REQUESTED** [review]

**Comment by Mahmoud-Emad** [comment]
image

can you please add some padding on the modal?

**Review by Mahmoud-Emad — APPROVED** [review]

**Pull request merged**